### PR TITLE
Get Ember tests to run; add them to Fabric/Travis

### LINKED ISSUE #166: Travis should run `ember test` on all builds
This will run the front-end tests and lint the js files. We already do this on the backend so it is a good practice.

### PR DESCRIPTION
**Pull request by ConnorCimowsky**
:information_source:  Resolves #166

### CODE REVIEW

**Comment by Kapin** [comment]
LGTM

**Comment by ConnorCimowsky** [comment]
This actually doesn't work as of now, I submitted the PR so I could see what happens with Travis. I'll let y'all know when it's ready for review, since it'll likely be more complicated than it currently is. :hankey:

**Comment by Kapin** [comment]
I just saw the travis changes now. If you need help poking around at it we can sit down and sort it out.

**Comment by ConnorCimowsky** [comment]
Cool, thanks! I think I might almost have it working, but if it doesn't we can pair on Monday

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,11 @@ addons:
 branches:
   only:
     - master
+before_install:
+  - "npm install -g npm@^2"
```
Comment (Kapin) [inline .travis.yml]: The node versioning syntax still makes absolutely no sense :confounded:

File: scripts/install-frontend.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+cd src-frontend
+npm install -g bower
+npm install
+bower install
```
Comment (Kapin) [inline scripts/install-frontend.sh]: nit, but can this go back to the previous directory when done? I feel like it won't matter all that much, but it is good to cleanup.

**Comment by Kapin** [comment]
Only question is does ember test run jshint as well? I think it's good practice for us to run it on every build as well.

**Comment by Kapin** [comment]
Also can you add this to the pre-commit hooks and the fab test command?

File: scripts/install-frontend.sh (hunk 2 above)
Comment (ConnorCimowsky) [inline scripts/install-frontend.sh]: That's actually why I made this a separate script for `bash -x`. It won't affect the environment of the caller.

**Comment by ConnorCimowsky** [comment]
Yes this runs JSHint :thumbs:

**Comment by ConnorCimowsky** [comment]
Already in `fab test` but I'll add it to the hook

File: scripts/install-frontend.sh (hunk 2 above)
Comment (Kapin) [inline scripts/install-frontend.sh]: perfect

**Comment by Kapin** [comment]
LGTM once the hook is in.

File: scripts/install-frontend.sh (hunk 2 above)
Comment (Ommy) [inline scripts/install-frontend.sh]: If I call this while within `scripts/` then won't `cd src-frontend` fail?

**Comment by Ommy** [comment]
LGTM after comments + hooks.

Also, would you mind squashing some of those commits?

**Comment by Trinovantes** [comment]
+1

File: scripts/install-frontend.sh (hunk 2 above)
Comment (ConnorCimowsky) [inline scripts/install-frontend.sh]: How do you mean? Is there a non-gross way of doing this?

**Comment by ConnorCimowsky** [comment]
The Ember tests run in `fab test` which runs in `fab verify`, which runs in the pre-push hook, so we're good. Will squash and merge.

**Pull request merged**